source/agora/utils/Log.d

```diff
@@ -31,86 +31,98 @@ import ocean.transition;
 import ocean.util.log.AppendConsole;
 import ocean.util.log.Appender;
 import ocean.util.log.Event;
-import ocean.util.log.Logger;
+import Ocean = ocean.util.log.Logger;
 
 import std.algorithm : min;
 import std.format;
 import std.stdio;
 import std.range : Cycle, cycle, isOutputRange, take, takeExactly;
 
-/// Insert a logger in the current scope, named log
-public template AddLogger (string moduleName = __MODULE__)
+/// nothrow wrapper around Ocean's Logger
+public struct Logger
 {
-    import Ocean = ocean.util.log.Logger;
-    import agora.common.Types;
+    private Ocean.Logger logger;
+
+    /// ctor
+    public this (string moduleName)
+    {
+        import core.memory;
+        this.logger = Ocean.Log.lookup(moduleName);
+        this.logger.buffer(new char[](16384));
+        GC.addRoot(cast(void*)this.logger);
+    }
 
-    /// nothrow wrapper around Ocean's Logger
-    public static struct Logger
+    /// dtor
+    public ~this ()
     {
-        private Ocean.Logger logger;
+        import core.memory;
+        GC.removeRoot(cast(void*)this.logger);
+    }
 
-        // workaround: weird CT bug in NetworkManager.d call:
-        // log.info("Discovery reached. {} peers connected.", this.peers.length);
-        // source/agora/network/NetworkManager.d(149,12): Error: no property info for type Logger
-        public void info (Args...) (Args args) @safe nothrow
+    // workaround: weird CT bug in NetworkManager.d call:
+    // log.info("Discovery reached. {} peers connected.", this.peers.length);
+    // source/agora/network/NetworkManager.d(149,12): Error: no property info for type Logger
+    public void info (Args...) (Args args) @safe nothrow
```

Comment:
This bug has been fixed no ?

Reply:
It's funny you mention this, because I seem to have other issues now too. Like calling `.fatal` fails.

I feel like `opDispatch` is a very hacky implementation in DMD..